src/util/kaldi-table-inl.h

```diff
@@ -248,49 +271,66 @@ template<class Holder>  class SequentialTableReaderScriptImpl:
   }
 
   virtual ~SequentialTableReaderScriptImpl() {
-    if (state_ == kError)
-      KALDI_ERR << "Reading script file failed: from scp "
-                << PrintableRxfilename(script_rxfilename_);
-    // If you don't want this exception to be thrown you can
-    // call Close() and check the status.
-    if (state_ == kLoadSucceeded)
-      holder_.Clear();
+    if (this->IsOpen() && !Close())
+      KALDI_ERR << "TableReader: reading script file failed: from scp "
+                      << PrintableRxfilename(script_rxfilename_);
   }
  private:
   bool LoadCurrent() {
     // Attempts to load object whose rxfilename is on the current scp line.
-    if (state_ != kHaveScpLine)
-      KALDI_ERR << "LoadCurrent() called at the wrong time.";
-    bool ans;
-    // note, NULL means it doesn't read the binary-mode header
-    if (Holder::IsReadInBinary()) {
-      ans = data_input_.Open(data_rxfilename_, NULL);
-    } else {
-      ans = data_input_.OpenTextMode(data_rxfilename_);
-    }
-    if (!ans) {
-      // May want to make this warning a VLOG at some point
-      KALDI_WARN << "Failed to open file "
-                 << PrintableRxfilename(data_rxfilename_);
-      state_ = kLoadFailed;
-      return false;
-    } else {
-      if (holder_.Read(data_input_.Stream())) {
-        state_ = kLoadSucceeded;
-        return true;
-      } else {  // holder_ will not contain data.
-        KALDI_WARN << "Failed to load object from "
+   
+    if (state_ != kObjectLoaded) {
+      if (state_ != kHaveScpLine)
+        KALDI_ERR << "LoadCurrent() called at the wrong time.";
+      bool ans;
+      // note, NULL means it doesn't read the binary-mode header
+      if (Holder::IsReadInBinary()) {
+        KALDI_LOG << data_rxfilename_;
+        ans = data_input_.Open(data_rxfilename_, NULL);
+      } else {
+        ans = data_input_.OpenTextMode(data_rxfilename_);
+      }
+      if (!ans) {
+        // May want to make this warning a VLOG at some point
+        KALDI_WARN << "Failed to open file "
                    << PrintableRxfilename(data_rxfilename_);
         state_ = kLoadFailed;
         return false;
+      } else {
+        if (holder_.Read(data_input_.Stream())) {
+          if (range_.empty()) {
+            state_ = kLoadSucceeded;
+            return true;
+          } // o.w. we'll go ahead to extract the partial object.
+        } else {  // holder_ will not contain data.
+          KALDI_WARN << "Failed to load object from "
+                     << PrintableRxfilename(data_rxfilename_);
+          state_ = kLoadFailed;
+          return false;
+        }
       }
     }
+    
```

Comment:
Plus that template will just crash except for class Matrix.  
You don't need to have it work for the empty string- it should just crash in that case.  If there is no range, that function shouldn't be called.